content/develop/tutorials/execution-flow/fragments/trigger-a-full-script-rerun-from-a-fragment.md

```diff
@@ -0,0 +1,530 @@
+---
+title: Trigger a full-script rerun from inside a fragment
+slug: /develop/tutorials/execution-flow/trigger-a-full-script-rerun-from-a-fragment
+---
+
+# Trigger a full-script rerun from inside a fragment
+
+Streamlit lets you turn functions into [fragments](/develop/concepts/architecture/fragments), which can rerun independently from the full script. When a user interacts with a widget inside a fragment, only the fragment ruruns. Sometimes, you may want to trigger a full-script rerun form inside a fragment. To do this, call [`st.rerun`](/develop/api-reference/execution-flow/st.rerun) inside the fragment.
+
+## Applied concepts
+
+- Use fragments to run a granular process separately from a related, coarser process.
```

Comment:
I get what you're saying but "coarser" feels a little weird here. Maybe "Use fragments to update part of your app, or update the full app, based on user input" or something a bit more explicit like that?

Reply:
"Use a fragment rerun part or all of your app, depending on user input."

Reply:
I think it should be "Use a fragment _TO_ rerun part or all of your app, depending on user input." No?